
syntax = "proto2";

package mesop.components.input;

// Next id: 26
message InputType {
  optional bool disabled = 1;
  optional string id = 2;
  optional string placeholder = 3;
  optional bool required = 5;
  optional string type = 6;
  optional string value = 8;
  optional bool readonly = 9;
  optional bool hide_required_marker = 10;
  optional string color = 11;
  optional string float_label = 12;
  optional string appearance = 13;
  optional string subscript_sizing = 14;
  optional string hint_label = 15;
  optional string label = 16;
  optional string on_input_handler_id = 17;
  optional string on_enter_handler_id = 24;
  optional string on_blur_handler_id = 25;
  // Used for textarea only.
  optional int32 rows = 18;
  optional bool autosize = 20;
  optional int32 min_rows = 21;
  optional int32 max_rows = 22;
  // Not exposed as public API.
  optional bool is_textarea = 19;
  optional bool is_native_textarea = 23;
}
